lib/demos/reference/motion_demo_shared_y_axis_transition.dart

```diff
@@ -0,0 +1,177 @@
+import 'dart:math';
+import 'package:flutter/material.dart';
+import 'package:animations/animations.dart';
+import 'package:gallery/l10n/gallery_localizations.dart';
+
+// BEGIN sharedYAxisTransitionDemo
+
+class SharedYAxisTransitionDemo extends StatefulWidget {
+  const SharedYAxisTransitionDemo({Key key}) : super(key: key);
+
+  @override
+  _SharedYAxisTransitionDemoState createState() =>
+      _SharedYAxisTransitionDemoState();
+}
+
+class _SharedYAxisTransitionDemoState extends State<SharedYAxisTransitionDemo>
+    with SingleTickerProviderStateMixin {
+  bool _isAlphabetical = false;
+  AnimationController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = AnimationController(
+      duration: const Duration(milliseconds: 300),
+      vsync: this,
+    );
+  }
+
+  final _recentList = ListView(
+    // Adding [UniqueKey] to make sure the widget rebuilds when transitioning.
+    key: UniqueKey(),
+    children: [for (int i = 0; i < 10; i++) _AlbumTile((i + 1).toString())],
```

Comment:
nice! Usually I see these split up into multiple lines. @rami-a what do you think?

Reply:
Agreed with that, something like

```suggestion
    children: [
      for (int i = 0; i < 10; i++)
         _AlbumTile((i + 1).toString()),
    ],
```